### ISSUE TITLE
[Arm64] Review/revise code in genAllocLclFrame

### ISSUE THREAD

**Issue by sdmaclea** [dotnet/coreclr#14113]
```
// TODO-ARM64-Bug?: set the availMask properly!
// TODO-LdStArch-Bug?: review this. The first time we load from [sp+0] which will always succeed. That doesn't make sense.
// TODO-ARM64-CQ: we could probably use ZR on ARM64 instead of rTemp.
```

Will this cause codegen on 64K page machine to fail to probe correctly on a 4K page machine?
`const size_t pageSize = compiler->eeGetPageSize();`

`if (frameSize >= 0x2000)` Magic number should be pageSize*2

category:implementation
theme:basic-cq
skill-level:intermediate
cost:small

**Comment by RussKeldorph** [comment]
> Will this cause codegen on 64K page machine to fail to probe correctly on a 4K page machine?

I would assume so, at least on Windows, though I doubt Windows will have more than one possible page size for a given architecture.  If it did, presumably we could use the "minimum" page size...either by having crossgen return a different result from eeGetPageSize or just adding a new API.

For Unix, I'm not sure the probes are required, so there may be no correctness issue even in scenarios where compile-time-page-size != run-time-page-size is possible.  If not, presumably another way to address this (and improve codegen) is to stop probing altogether.  C++ compilers apparently don't insert probes, and stack overflow is detected via the combination of SP location and SIGSEGV, so I'm not sure what value the probing has on Unix.  I haven't done much investigation, though, other than a brief conversation with @janvorli.  I could be missing something.

@dotnet/jit-contrib

**Comment by janvorli** [comment]
As an interesting coincidence, I was helping @rartemev today to debug a test crashing on RHEL 6. The interesting part is that sigsegv handler was called when we did a stack probe, but the location was in the middle of the stack range. When we have disabled the stack probes by commenting out the piece of jit code that inserts them in CodeGen::genAllocLclFrame, the test have passed ok. 
Is is known that while Linux reserves fixed range of virtual memory for the stack, it doesn't commit it all, but rather does it by chunks and so when it cannot access memory in the stack range, the kernel handles that access violation and commits another block of the stack and the process doesn't know about it.
But the RHEL 6 has an ancient kernel and our way of probing makes it think it is a regular sigsegv and so it calls the sigsegv handler in the process. 
We have done various experiments to find out the reason and we have finally found the cuprit. 
The reason is that the SP is above the probing addresses. If we modified the generated probing code so that we moved the SP down before the probing and it was always below the address we have probed, the issue went away.
So after seeing this, I think that we should really get rid of the probes on Unix. Since we fail fast no matter whether the stack overflow was detected by a probe or we just overflowed the stack without any probing (recursive function with small stack frame below the probing size threshold would do that), the probes don't give us any advantage.
